src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.StandardProperties.cs

```diff
@@ -164,7 +163,44 @@ public int KeySize
         {
             get
             {
-                int keySize = _keyHandle.GetPropertyAsDword(KeyPropertyName.Length, CngPropertyOptions.None);
+                int keySize = 0;
+                int cbResult;
+
+                unsafe
+                {
+                    // Attempt to use PublicKeyLength first as it returns the correct value for ECC keys
+                    ErrorCode errorCode = Interop.NCrypt.NCryptGetProperty(
+                        _keyHandle,
+                        KeyPropertyName.PublicKeyLength,
+                        &keySize,
```

Comment:
I wonder if we've done this enough now to warrant making a NCryptGetIntProperty which is out the int value, and passes sizeof(int) as a fixed thing.

Just to keep the unsafe code limited to where we really need it.